addons/iap/models/iap_account.py

```diff
@@ -63,13 +63,11 @@ class IapAccount(models.Model):
         help="The In-App Purchase pack the system will automatically buy when reaching your refill threshold.",
     )
 
-    @api.depends('balance')
-    def _compute_balance_amount(self):
+    @api.depends('balance_amount')
```

Comment:
It should also depend on `service_id.unit_name` and `service_id.integer_balance`